runtime/src/main/java/io/quarkiverse/messaginghub/pooled/jms/PooledJmsDecorator.java

```diff
@@ -30,7 +35,13 @@ public PooledJmsDecorator(PooledJmsRuntimeConfig config) {
 
     private ConnectionFactory getConnectionFactory() {
         if (factory == null) {
-            factory = wrapper.wrapConnectionFactory(delegate);
+            Object subClass = Reflections.readField(delegate.getClass(), "subclass", delegate);
```

Comment:
Why do you read the field with reflection? Where does the `subclass` field come from?

Reply:
And where's the `subclass` field defined?

Reply:
TBH, I just find this field by using IntelliJ IDEA Debuger. It seems  in a generated class by Quarkus, such like `io.quarkiverse.messaginghub.pooled.jms.it.PooledJmsDecorator_050a96e281a21f769caecc114a7900327ea6ddfc_Delegate_Subclass`

Reply:
I see. 

Well, you should definitely not rely on the presence of this field because it's an implementation detail, i.e. it does not have to be generated in future versions of quarkus.

I need to check the code but in general, the delegate object injected into a `@Delegate` injection must only  implement the delegate type, and it makes sense because the decorator does not really know what bean is being decorated.

Reply:
Yes, is it possible to unwrap such delegate object to get the original bean?

Reply:
I'm not aware of any way how to do this. In any case, it feels wrong to me that a decorator for `ConnectionFactory` expects the decorated bean to also implement `XAConnectionFactory`. I understand that it might be the case you're trying to fix but it's not correct in general :shrug:.

Reply:
@mkouba - Thanks a lot for the clarifying!

Reply:
@mkouba Hmm, how about to introduce a composed interface like

```
public interface ComposedConnectionFactory extends ConnectionFactory, XAConnectionFactory {
}
```

Reply:
That should work but the decorated bean would need to implement this interface ofc.

Reply:
And the bean could be like
```
@Dependent
public class IBMMQConnectionFactory extends MQXAConnectionFactory implement ComposedConnectionFactory {
...
}
```

I just test and it works. So what do you think?

Reply:
Yes, something like this would be decorated by a decorator that has `@Inject @Any @Delegate ComposedConnectionFactory delegate;`... and the injected delegate would implement `ComposedConnectionFactory` (and thus also `XAConnectionFactory`).

Reply:
Nice, I will prepare the other PR.